cmd/oras/internal/errors/errors.go

```diff
@@ -60,6 +65,45 @@ func CheckArgs(checker func(args []string) (bool, string), Usage string) cobra.P
 	}
 }
 
+// Handler handles error during cmd execution.
+type Handler interface {
+	// Handle handles error during cmd execution.
+	// if nil is returned, then err cannot be handled.
+	Handle(err error, cmd *cobra.Command) error
+}
+
+// Command returns an error-handled cobra command.
+func Command(cmd *cobra.Command, handler Handler) *cobra.Command {
+	runE := cmd.RunE
+	cmd.RunE = func(cmd *cobra.Command, args []string) error {
+		err := runE(cmd, args)
+		if err == nil {
+			return nil
+		}
+		if handled := handler.Handle(err, cmd); handled != nil {
+			return handled
+		}
+		return err
+	}
+	return cmd
+}
+
+// GetInner gets the inner error from the error response.
+func GetInner(err error, errResp *errcode.ErrorResponse) error {
+	inner := errResp.Errors
+	if len(inner) == 0 {
+		return fmt.Errorf("empty response body: %w", errResp)
```

Comment:
`len(inner) == 0` does not imply empty response body but unrecognised response body.

See code: https://github.com/mostoja/oras-go/blob/9cac41930202c5f5fd8f5a2041862d593571c564/registry/remote/internal/errutil/errutil.go#L33-L48

Reply:
What do you suggest as a proper output? 
How about `no error code found: `? 
I suggested to change the prefix to `Empty error response from registry` but rejected by @FeynmanZhou. He thinks it's too complicated for user to know three different kind of prefixes.